storage/remote/queue_manager.go

```diff
@@ -514,8 +530,14 @@ func NewQueueManager(
 	return t
 }
 
-// AppendMetadata sends metadata to the remote storage. Metadata is sent in batches, but is not parallelized.
-func (t *QueueManager) AppendMetadata(ctx context.Context, metadata []scrape.MetricMetadata) {
+// AppendWatcherMetadata sends metadata to the remote storage. Metadata is sent in batches, but is not parallelized.
+// This is only used for the 1.x proto format.
+func (t *QueueManager) AppendWatcherMetadata(ctx context.Context, metadata []scrape.MetricMetadata) {
+	// no op for any newer proto format, which will cache metadata sent to it from the WAL watcher.
+	if t.rwFormat > MinStrings {
```

Comment:
Should we kill this feature (in separate PR I guess?)

Reply:
Maybe? 

We'd essentially be moving, or duplicating? the caching from the scrape code so we should double check that. It would be nice to clean up, then we'd just have code in queue_manager that starts up a routine and handles sending metadata in a batch from the cache if using 1.x, otherwise we shove metadata into the new 2.0 proto format when we read a record in the WAL.

It would be nice to build it so that we have a cache for the metadata that can be shared by all queue_managers and avoid another "labels cache per queue manager" situation. The hard part there is how do we know when we can remove something from the cache, given that all queue_managers could be at different points in the WAL.